documentation/attributes.html

```diff
@@ -41,47 +26,471 @@
+    writefilter.attribute.enumstorelimit</a>
+  </td><td>content.proton.documentdb.attribute.resource_usage.enum_store</td>
+  <td>For string attribute fields or attribute fields with
+    <a href="#fast-search">fast-search</a>,
+    there is a max limit on the number of unique values that can be stored for that attribute.
+    The limit is approx 2 billion.
+    The component storing these values is called enum store</td></tr>
+<tr><td>attribute multivalue</td>
+  <td><a href="https://github.com/vespa-engine/vespa/blob/master/searchcore/src/vespa/searchcore/config/proton.def">
+    writefilter.attribute.multivaluelimit</a>
+  </td><td>content.proton.documentdb.attribute.resource_usage.multi_value</td>
+  <td>For array or weighted set attribute fields,
+    there is a max limit on the number of documents that can have the same number of values.
+    The limit is 2 billion documents per node</td></tr>
+</tbody>
+</table>
+An error is emitted when exceeding the limit - sample:
+<pre>
+Detail resultType=FATAL_ERRORexception=
+'ReturnCode(NO_SPACE, Put operation rejected for document 'id:mynamespace:mydoc::123456' of type 'mydoc':
+'enumStoreLimitReached: {
+  action: "add more content nodes",
+  reason: "enum store address space used (0.92813) &gt; limit (0.9)",
+  enumStore: { used: 31890298144, dead: 0, limit: 34359738368},
+  attributeName: "text", subdb: "ready"}')'
+endpoint=vespa1:8080 ssl=false resultTimeLocally=1532685239428
+</pre>
+A similar message is emitted for too many values for a multivalue attribute.
+</p><p>
+  To fix a problem with too many values,
+  add content nodes to distribute documents with attributes over more nodes.
+</p><p>
+  Use metrics <em>content.proton.documentdb.attribute.resource_usage.enum_store.average</em> and
+  <em>content.proton.documentdb.attribute.resource_usage.multi_value.average</em>
+  to track usage.
 </p>
 
 
 
-<h2 id="sizing-attribute-memory">Sizing attribute memory</h2>
+<h2 id="document-meta-store">Document meta store (implicit attribute)</h2>
 <p>
-Attribute data can consume a lot of memory, especially when using <em>fast-access</em>,
-where <em>not ready</em> copies are brought into memory.
-Please see <a href="performance/attribute-memory-usage.html">attribute memory usage sizing</a>.
+  The document meta store is an in-memory data structure used for bookkeeping about every
+  document stored on a node.
+</p><p>
+  The document meta store is an implicit attribute, and is
+  <a href="proton.html#lid-space-compaction">compacted</a> and
+  <a href="proton.html#attribute-flush">flushed</a>.
+  Memory usage for applications with small documents can be dominated by this attribute,
+  particularly for <em>store-only</em> and <em>streaming search</em> applications.
+</p><p>
+  The document meta store scales linearly with number of documents -
+  using approximately 30 bytes per document on disk.
```

Comment:
Consider removing "on disk", as this can be interpreted as the document meta store uses 30 bytes (on disk instead of memory) per document.